pypeit/core/wave.py

```diff
@@ -309,44 +313,92 @@ def flexure_obj(specobjs, maskslits, method, sky_file, mxshft=None):
                 if sv_fdict is not None:
                     msgs.warn("Will used saved estimate from a previous slit/object")
                     fdict = copy.deepcopy(sv_fdict)
                 else:
-                    msgs.warn("No previous good solution.  Punting on this object")
-                    continue
+                    # One does not exist yet
+                    # Save it for later
+                    return_later_sobjs.add([slit, ss])
+                    punt = True
             else:
                 sv_fdict = copy.deepcopy(fdict)
 
-            # Simple interpolation to apply
-            npix = len(sky_wave)
-            x = np.linspace(0., 1., npix)
-            # Apply
-            for attr in ['boxcar', 'optimal']:
-                if not hasattr(specobj, attr):
-                    continue
-                if 'WAVE' in getattr(specobj, attr).keys():
-                    msgs.info("Applying flexure correction to {0:s} extraction for object:".format(attr) +
-                              msgs.newline() + "{0:s}".format(str(specobj)))
-                    f = interpolate.interp1d(x, sky_wave, bounds_error=False, fill_value="extrapolate")
-                    getattr(specobj, attr)['WAVE'] = f(x+fdict['shift']/(npix-1))*units.AA
-            # Shift sky spec too
-            cut_sky = fdict['sky_spec']
-            x = np.linspace(0., 1., cut_sky.npix)
-            f = interpolate.interp1d(x, cut_sky.wavelength.value, bounds_error=False, fill_value="extrapolate")
-            twave = f(x + fdict['shift']/(cut_sky.npix-1))*units.AA
-            new_sky = xspectrum1d.XSpectrum1D.from_tuple((twave, cut_sky.flux))
+            # Punt?
+            if punt:
+                break
 
+            # Interpolate
+            new_sky = flexure_interp(sky_wave, specobj, fdict)
             # Update dict
             for key in ['polyfit', 'shift', 'subpix', 'corr', 'corr_cen', 'smooth', 'arx_spec']:
                 flex_dict[key].append(fdict[key])
             flex_dict['sky_spec'].append(new_sky)
+
         flex_list.append(flex_dict.copy())
+
+        # Do we need to go back?
+        for items in return_later_sobjs:
+            if sv_fdict is None:
+                msgs.info("No flexure corrections could be made")
+                break
+            # Setup
+            slit, ss = items
+            flex_dict = flex_list[slit]
+            specobj = specobjs[ss]
+            sky_wave = specobj.boxcar['WAVE'] #.to('AA').value
+            # Copy me
+            fdict = copy.deepcopy(sv_fdict)
+            # Interpolate
+            new_sky = flexure_interp(sky_wave, specobj, fdict)
+            # Update dict
+            for key in ['polyfit', 'shift', 'subpix', 'corr', 'corr_cen', 'smooth', 'arx_spec']:
+                flex_dict[key].append(fdict[key])
+            flex_dict['sky_spec'].append(new_sky)
+
     return flex_list
 
 
+def flexure_interp(sky_wave, specobj, fdict):
```

Comment:
I have some coding practice issues with this function.  Beyond that though, would this be better placed as a method in the `SpecObj` class?

Reply:
Good call.  Have shoved it there.

I don't like this function either.
I simply moved it from the previous...